internal/backend/db/dbgorm/envs.go

```diff
@@ -43,7 +43,12 @@ func (gdb *gormdb) deleteEnvs(ctx context.Context, ids []sdktypes.UUID) error {
 		return err
 	}
 
-	return db.Where("env_id IN ?", ids).Delete(&scheme.Env{}).Error
+	// delete envs and associated vars
+	if err := db.Where("env_id IN ?", ids).Delete(&scheme.Env{}).Error; err != nil {
+		return err
+	}
+
+	return db.Where("var_id IN ?", ids).Delete(&scheme.Var{}).Error
```

Comment:
1. I don't think I could delete from two different tables in one statement
2. we also cannot use on delete cascade - (i) gorm not supporting it on soft-delete (env) and (ii) env isn't a foreign key and cannot be since it's a scope id - either var or connection

Reply:
No, I meant this: https://gorm.io/docs/transactions.html

This allows you to run multiple queries as a single unit, so it one fails they all fail together. This is done to avoid inconsistent states and having to implement custom rollback logic.

Anyway, if this makes sense, you can do it in a separate PR.